packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.test.ts

```diff
@@ -2,178 +2,67 @@
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
-
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName ON JoinClassName.propName1='someName'");
-    });
-
-    it("should return a category query string, when property is category", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
+    sut = new QueryBuilder(dataProvider);
+  });
 
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: true,
-            value: "propValue1",
-          },
-        ],
-      };
+  afterEach(() => {
+    dataProvider.getFieldByPropertyRecord.reset();
+  });
 
-      const query: Query = {
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
+  const queryBuildertestData: QueryBuilderTestData = testData;
+  queryBuildertestData.testCases.forEach((testCase) => {
+    it(testCase.name, async () => executeTest(
+      sut,
+      testCase.expectedResult,
+      testCase.operations,
+    ));
+  });
 
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName JOIN bis.Category ON bis.Category.ECInstanceId = bis.GeometricElement3d.category.id AND ((bis.Category.CodeValue='propValue1') OR (bis.Category.UserLabel='propValue1'))");
-    });
+  const mockPropertiesField = (propertyRecord: PropertyRecord, propertiesField: PropertiesField) => {
+    const propertiesFieldMock: StubbedType<PropertiesField> = MockFactory.create(PropertiesField);
 
-    it("should return query string with where clause, when base class has properties", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue1",
-          },
-          {
-            name: "propName2",
-            needsQuote: false,
-            isCategory: false,
-            value: "propValue2",
-          },
-        ],
-      };
+    MockFactory.stubProperty(propertiesFieldMock, "parent", () => propertiesField.parent);
+    MockFactory.stubProperty(propertiesFieldMock, "properties", () => propertiesField.properties);
+    MockFactory.stubProperty(propertiesFieldMock, "type", () => propertiesField.type);
 
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
+    dataProvider.getFieldByPropertyRecord.withArgs(propertyRecord).returns(Promise.resolve(propertiesFieldMock));
```

Comment:
`.resolves(propertiesFieldMock)` instead of `returns(Promise.resolve(...`

Reply:
Changed

Reply:
Resolved